glean-core/ios/Glean/Scheduler/MetricsPingScheduler.swift

```diff
@@ -201,13 +197,10 @@ class MetricsPingScheduler {
             //
             // During the Glean initialization, we require any metric recording to be
             // batched up and replayed after any startup metrics ping is sent. To guarantee
-            // that, we dispatch this function from `Dispatchers.API.executeTask`. However,
-            // Pings.metrics.submit() ends up calling `Dispatchers.API.launch` again which
-            // will delay the ping collection task after any pending metric recording is
-            // executed, breaking the 'metrics' ping promise of sending a startup 'metrics'
-            // ping only containing data from the previous session.
-            // To prevent that, we synchronously manually dispatch the 'metrics' ping, without
-            // going through our public API.
+            // that we synchronously and manually dispatch the 'metrics' ping without
+            // going through our public API.  Since this part of the `Glean.initialize`
```

Comment:
I'm not super happy to tie this so closely to where it  is called in initialize.
I'd rather phrase this in terms of `startupPing`, because a call with `startupPing = true` should only ever happen at most once per application run.